### PR TITLE
Python Import Style Guide Changes

### PR DESCRIPTION
**Pull request by krak3n**
This PR updates our python import style guides to follow the Google Import organisation which breaks imports out into 3 groups:

* Standard Lib
* Third Party
* Application Level

I've also added Future imports to the top of this list.

### CODE REVIEW

**Comment by krak3n** [comment]
FYI no need to change existing stuff straight away, just as we go.

File: styleguides/python.md [hunk 1]
```diff
@@ -81,8 +92,17 @@ import time
 from dateutil.parser import parse
 from dateutil.tz import tzutc
 from safelogger import getMultiProcessLogger
+
+import requests
+
```
Comment (radeklos) [inline styleguides/python.md]: There shouldn't be empty line..

File: styleguides/python.md [hunk 2]
```diff
@@ -81,8 +92,17 @@ import time
...
+from thirdparty.library import something
+
+from application.http import OK
+from application.models import Foo
 ```
 
+Note how ``import`` and ``from`` statements are always sperated by new lines and how the imports are grouped.
```
Comment (radeklos) [inline styleguides/python.md]: separated?

**Comment by radeklos** [comment]
Isort should do the think: https://pypi.python.org/pypi/isort. I've been using this script to sort all imports in files that I've changed: https://gist.github.com/radeklos/437f856e30e36ccc8820

**Comment by krak3n** [comment]
@radeklos looks cool and has vim plugin: https://github.com/fisadev/vim-isort

**Comment by radeklos** [comment]
There is also a plugin for sublime but not sure if we can really use it because we keep all dependencies in containers.

**Comment by krak3n** [comment]
@radeklos 6526cfd should fix your comments

**Comment by radeklos** [comment]
:+1:

**Comment by peeklondon** [comment]
:+1: nice

**Comment by krak3n** [comment]
@radeklos just playing with `isort`. We could place a `.isort.cfg` in each projects repository with standard settings. We can force unknown packages which would be third party into the third party section and define known first party libs to ensure third and first party libs are broken out correctly, like this:
 
``` cfg
[settings]
line_length=80
multi_line_output=3
known_first_party=fm
default_section=THIRDPARTY
import_heading_stdlib=Standard Libs
import_heading_thirdparty=Third Party Libs
import_heading_firstparty=First Party Libs
```

Results in:

``` python
# Standard Libs
from functools import wraps

# Third Party Libs
from flask import g, has_request_context, request
from itsdangerous import URLSafeTimedSerializer
from werkzeug import LocalProxy

# First Party Libs
from fm.ext import config, redis
from fm.http import Unauthorized
from fm.models.user import User
```

Then we can use editor tools to sort imports for us which I'm all for so import organisation is standardised .

**Comment by radeklos** [comment]
It sounds good for me..

**Pull request merged**